src/dataset-builder/DatasetBuilder.ts

```diff
@@ -481,15 +483,20 @@ const RequestAccessModal = (props: RequestAccessModalProps) => {
           onClick: async () => {
             await DataRepo()
               .dataset(datasetId)
-              .createSnapshotRequest({
-                name,
-                researchPurposeStatement,
-                datasetRequest: {
-                  cohorts,
-                  conceptSets,
-                  valueSets: _.map((valuesSet) => ({ domain: valuesSet.header, values: valuesSet.values }), valuesSets),
-                },
-              });
+              .createSnapshotRequest(
+                convertDatasetAccessRequest({
```

Comment:
This code is creating a object that's only used here only to "convert" it to another type. Why is the intermediate type needed? I would instead write a `create` function that creates the type directly, rather than converting.